src/blue.ts

```diff
@@ -109,20 +115,30 @@ export function blueProxyFactory(env: MembraneBroker) {
         return redPartialDesc;
     }
 
+    function copyRedDescriptorIntoShadowTarget(shadowTarget: BlueShadowTarget, originalTarget: BlueProxyTarget, key: PropertyKey) {
+        // Note: a property might get defined multiple times in the shadowTarget
+        //       but it will always be compatible with the previous descriptor
+        //       to preserve the object invariants, which makes these lines safe.
+        const normalizedRedDescriptor = ReflectGetOwnPropertyDescriptor(originalTarget, key);
+        if (!isUndefined(normalizedRedDescriptor)) {
+            const blueDesc = getBlueDescriptor(normalizedRedDescriptor);
+            ReflectDefineProperty(shadowTarget, key, blueDesc);
+        }
+    }
+
     function lockShadowTarget(shadowTarget: BlueShadowTarget, originalTarget: BlueProxyTarget) {
         const targetKeys = ownKeys(originalTarget);
         for (let i = 0, len = targetKeys.length; i < len; i += 1) {
-            const key = targetKeys[i];
-            const blueDesc = ReflectGetOwnPropertyDescriptor(shadowTarget, key);
-            if (isUndefined(blueDesc) || blueDesc.configurable === true) {
-                const redDesc = ReflectGetOwnPropertyDescriptor(originalTarget, key) as PropertyDescriptor;
-                ReflectDefineProperty(shadowTarget, key, getBlueDescriptor(redDesc));
-            }
+            copyRedDescriptorIntoShadowTarget(shadowTarget, originalTarget, targetKeys[i]);
         }
         ReflectPreventExtensions(shadowTarget);
     }
 
-    class BlueProxyHandler implements ProxyHandler<BlueProxyTarget> {
+    function getStaticRedArray(blueArray: BlueArray): RedArray {
+        return map(blueArray, (blue: BlueValue) => env.getRedValue(blue));
```

Comment:
If `env.getRedValue` accepts just 1 arg you can do:
```js
 map(blueArray, env.getRedValue)
```

Reply:
will do